src/packages/@ncigdc/components/LoadableWithLoading/Loading.js

```diff
@@ -0,0 +1,31 @@
+import React from 'react';
+import Overlay from '@ncigdc/uikit/Overlay';
+import Spinner from '@ncigdc/uikit/Loaders/Material';
+
+const Loading = ({ isLoading, timedOut, pastDelay, error }) => {
+  if (isLoading) {
+    // While our other component is loading...
+    if (timedOut) {
+      // In case we've timed out loading our other component.
+      return <div>Loader timed out!</div>;
+    } else if (pastDelay) {
+      // Display a loading screen after a set delay.
+      return (
+        <Overlay show={true} style={{ position: 'absolute', zIndex: 10 }}>
+          <Spinner />
+        </Overlay>
+      );
+    } else {
+      // Don't flash "Loading..." when we don't need to.
+      return null;
+    }
+  } else if (error) {
+    // If we aren't loading, maybe
+    return <div>Error! Component failed to load</div>;
+  } else {
+    // This case shouldn't happen... but we'll return null anyways.
+    return null;
+  }
+};
```

Comment:
It would be nice to have unit tests for this function, seems really important as its wrapping all the routes.

Reply:
Since this PR is already merged, could you add a ticket for the unit test?

Reply:
https://jira.opensciencedatacloud.org/browse/PRTL-1797